### DISCUSSION TITLE
Change steam online status (e.g. Online)

### DISCUSSION THREAD

**Discussion by tmontney** [ValvePython/steam#421] [Q&A] [answered]
https://steam.readthedocs.io/en/latest/api/steam.client.builtins.html?highlight=persona#steam.client.builtins.user.User.change_status

I'm looking to achieve the same as going into the Steam desktop client > Friends > Online.

```python
self.client = SteamClient()
self.client.cli_login(username, password)
self.client.change_status(persona_state=EPersonaState.Online)
```

No errors, just doesn't change anything on the Steam desktop client. `self.client.persona_state` returns the correct state.

**Comment by tmontney** [comment]
I'm going to assume this is because I'm creating a new steam client, not connecting to my existing one. Is this project capable of that?

**Comment by rossengeorgiev** [comment, marked as answer]
When you call `self.client.change_status(...)` a message is queued to be sent. You need to yield to the `gevent` event loop to allow the message to be processed. You could use `client.idle()` or `client.sleep(seconds)`. That would change the online status of `client` instance, which a separate session for your account that the session on Steam desktop. If your status changes in one session, it may display in other sessions, like when you are playing a game, etc. Once you kill that session, it will revert back.

Note that you can't use this to control you Steam desktop client. For example, if you want to create a script change your status.
